sale_stock_available_to_promise_release/models/sale_order_line.py

```diff
@@ -1,13 +1,87 @@
 # Copyright 2019 Camptocamp (https://www.camptocamp.com)
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
-from odoo import models
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    def _prepare_procurement_values(self, group_id=False):
-        values = super()._prepare_procurement_values(group_id)
-        values["date_priority"] = self.order_id.date_order
-        return values
+    availability_status = fields.Selection(
+        selection=[
+            ("on_order", "On order"),
+            ("full", "Fully Available"),
+            ("partial", "Partially Available"),
+            ("restock", "Restock ordered"),
+            ("no", "Not available"),
+        ],
+        compute="_compute_availability_status",
+    )
+    expected_availability_date = fields.Datetime(compute="_compute_availability_status")
+    available_qty = fields.Float(
+        digits="Product Unit of Measure", compute="_compute_availability_status"
+    )
+
+    def _on_order_route(self):
+        self.ensure_one()
+        mto_route = self.env.ref("stock.route_warehouse0_mto")
+        product_is_mto = mto_route in self.product_id.route_ids
+        line_is_mto = mto_route == self.route_id
+        return product_is_mto or line_is_mto
+
+    @api.depends(
+        "move_ids.ordered_available_to_promise_uom_qty",
+        "product_id.route_ids",
+        "route_id",
+        "product_uom_qty",
+        "display_type",
+        "is_delivery",
+    )
+    def _compute_availability_status(self):
+        for record in self:
+            if record.display_type or not record.product_id:
```

Comment:
not a blocker but I would split this in smaller methods

Reply:
it's not really smaller, but I put the computation of the data in its own method to ease the override.